src/main/kotlin/it/pagopa/ecommerce/helpdesk/services/v1/HelpdeskService.kt

```diff
@@ -168,6 +177,179 @@ class HelpdeskService(
+                                }
+                            }
+                            ?.toList()
+                }
+            }
+            .doOnNext { orderResponse ->
+                logger.info(
+                    "Retrieved NPG operations for transaction [{}]: found {} operations",
+                    transactionId,
+                    orderResponse.operations?.size ?: 0
+                )
+            }
+    }
+
+    fun retrieveTransactionDetails(
+        transactionId: String
+    ): Mono<NTuple4<String, String, String, PaymentMethod>> {
+        return Mono.just(transactionId)
+            .flatMapMany {
+                ecommerceTransactionDataProvider
+                    .getTransactionsEventStoreRepository()
+                    .findByTransactionIdOrderByCreationDateAsc(transactionId)
+            }
+            .reduce(
+                it.pagopa.ecommerce.commons.domain.v2.EmptyTransaction(),
+                it.pagopa.ecommerce.commons.domain.v2.Transaction::applyEvent
+            )
+            .cast(BaseTransaction::class.java)
+            .map { baseTransaction ->
+                val transactionActivatedData =
+                    if (baseTransaction is BaseTransactionWithPaymentToken) {
+                        baseTransaction.transactionActivatedData
+                    } else null
+
+                val authRequestData =
+                    when (baseTransaction) {
+                        is BaseTransactionWithRequestedAuthorization ->
+                            baseTransaction.transactionAuthorizationRequestData
+                        else -> null
+                    }
+
+                val correlationId =
+                    (transactionActivatedData?.transactionGatewayActivationData
+                            as? NpgTransactionGatewayActivationData) // also contains orderId
+                        ?.correlationId
+                        ?: throw NoOperationDataFoundException(
+                            "No correlation ID found for transaction $transactionId"
+                        )
+                NTuple4(
+                    authRequestData?.authorizationRequestId
+                        ?: throw NoOperationDataFoundException(
+                            "No authorization request ID found for transaction $transactionId"
+                        ),
+                    authRequestData.pspId
+                        ?: throw NoOperationDataFoundException(
+                            "No PSP ID found for transaction $transactionId"
+                        ),
+                    correlationId,
+                    PaymentMethod.valueOf(
+                        authRequestData.paymentMethodName
+                            ?: throw NoOperationDataFoundException(
+                                "No payment method found for transaction $transactionId"
+                            )
+                    )
+                )
+            }
+    }
```

Comment:
Can we move this method to the dataprovider?

Reply:
And then we can remove this:
https://github.com/pagopa/pagopa-ecommerce-helpdesk-service/blob/765ac3049a0b9ea03f93d641aae57879f774a29f/src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v1/mongo/EcommerceTransactionDataProvider.kt#L190-L192

Reply:
Sure, reworked with [57387fb](https://github.com/pagopa/pagopa-ecommerce-helpdesk-service/pull/156/commits/57387fba33f58056ba2afa1b51309b7abd0bdbf3)